### PR TITLE
Event type parity, style and text copy updates

### LINKED ISSUE #1040: Design: Event notification naming scheme / layout
Suggestions were made re: event notifications in the realm of:

- {Noun}:{Verb}

- {Linode Label}:{Event Action}

Regarding time stamp, up for discussion. Please post comments if other ideas.

### LINKED ISSUE #1056: Notification behavior update
* Hover on entire notification
* Clicking on notification marks read and hides dropdown and navigates to the page
* No transition / delay on notification dropdown

### LINKED ISSUE #1086: Events API parity

### PR DESCRIPTION
**Pull request by mparke**
closes #1040 
closes #1086 
closes #1056 

- breaks `Notifications.js` into 2 files, one for the list, and one for the item
- adds an `EventTypeMap` ( in favor of extending switch statements ) defining all event types for api parity ( in it's own file )
- each list item is a `<Link>` that redirects to a corresponding page defined in the `EventTypeMap`
- dispatch calls mostly pulled out of the view layer
- some now-dead scss removed
- `font-xs-normal` decreased in size by `0.1 rem`
- TODOs added for:
    - `rate` display
    - `user_id` display

screen shot 2017-01-26 at 6 40 58 pm

Sorry the pull is pretty large, will comment for clarity where needed. 

Still needs test updates and linting.

### CODE REVIEW

**Comment by mparke** [comment]
realized this also closes #1056 so I added it to description.

File: scss/components/Notifications.scss [hunk 1]
```diff
@@ -1,105 +1,53 @@
...
-        height: 100%;
-        width: 100%;
-        left: 0;
-        top: $nav-height - 1;
-        background: $transparent;
-        z-index: 1000;
+        right: 0;
+        top: $nav-height;
+        width: $notifications-width;
+        z-index: 1001;
```
Comment (eatonphil) [inline scss/components/Notifications.scss]: Could you add a TODO here to make a z-index section in the variables file (or just start one)

File: scss/components/Notifications.scss [hunk 2]
```diff
@@ -1,105 +1,53 @@
...
+        &:focus {
+            color: $black;
+            text-decoration: none;
+            outline: none;
         }
 
-        .fa {
-            padding-left: 5px;
-            color: $blue;
+        &:hover,
```
Comment (eatonphil) [inline scss/components/Notifications.scss]: I think you can drop the & here

File: src/components/notifications/EventTypes.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    presentTenseAction: 'Resetting password for',
+    pastTenseAction: 'reset',
+    pastTensePrefix: 'Password for',
+    redirectUrl: getLinodeRedirectUrl,
+  },
+
+  dns_zone_create: {
+    presentTenseAction: 'Creating',
+    pastTenseAction: 'created',
+    redirectUrl: baseRedirect,
```
Comment (eatonphil) [inline src/components/notifications/EventTypes.js]: TODO?

File: scss/components/Notifications.scss (hunk 2 above)
Comment (mparke) [inline scss/components/Notifications.scss]: tried it, does not have the same result

File: src/components/notifications/EventTypes.js (hunk 3 above)
Comment (mparke) [inline src/components/notifications/EventTypes.js]: yeah, good call, we're going to need to address these, I could drop TODOs in or create an issue for each type missing redirects?

**Comment by mparke** [comment]
@eatonphil removed bold text for read events, keeps sorting method for now and reverses results

**Comment by eatonphil** [comment]
Cool, we'll find another reviewer Monday.

**Pull request merged**